evaluation/prompt_eval.ipynb

```diff
@@ -0,0 +1,552 @@
+              "5  The requested information is not available in ...          15.426303   \n",
+              "\n",
+              "                                          Evaluation  \n",
+              "0  The response provided does not accurately refl...  \n",
+              "1  The previous response does not accurately refl...  \n",
+              "2  The response provided in the previous message ...  \n",
+              "3  The response provided is accurate and adheres ...  \n",
+              "4  The response provided does not accurately refl...  \n",
+              "5  The response provided is out-of-domain as it d...  "
+            ]
+          },
+          "execution_count": 8,
+          "metadata": {},
+          "output_type": "execute_result"
+        }
+      ],
+      "source": [
+        "output_dataset"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 9,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "c6gvobzrHKnm",
+        "outputId": "f5c83923-4302-4902-a1ed-57b4f7f90866"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "Processing complete. Results saved to output_data.xlsx\n"
+          ]
+        }
+      ],
+      "source": [
+        "print(f\"Processing complete. Results saved to {output_file_path}\")"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 10,
+      "metadata": {},
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "Question 0 processed succesfully\n",
+            "Question 1 processed succesfully\n",
+            "Question 2 processed succesfully\n",
+            "Question 3 processed succesfully\n",
+            "Question 4 processed succesfully\n",
+            "Question 5 processed succesfully\n"
+          ]
+        }
+      ],
+      "source": [
+        "\"\"\" Compare response Evaluation \"\"\"\n",
+        "results = []\n",
+        "for index, row in dataset.iterrows():\n",
+        "    try:\n",
+        "        if index >= limit:\n",
+        "            break\n",
+        "        prompt = row[\"Prompt\"]\n",
+        "        evaluate_response = evaluate_and_compare_prompt_responses(prompt)\n",
+        "        results.append(evaluate_response)\n",
+        "        print(f\"Question {index} processed succesfully\")\n",
+        "    except Exception as e:\n",
+        "        print(f\"Error Question {index} processed unsuccesfully: {e}\")\n",
+        "        continue"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 11,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "comparison_output_dataset = pd.DataFrame(results, columns=['Question', 'Answer 1', 'Answer 2', 'Response Time 1 (s)', 'Response Time 2 (s)', 'Evaluation',])"
```

Comment:
What is the expectation from this comparison? There is now change in the system between the first run and this one, right ? So what does this result tell us ?

Reply:
I ran with a few assumptions that in each prompt/request would generate a different response in prod, I will however restructure the script to run comparisons based on the previous run and not within the same session/run.

Reply:
Same session/run isn't necessarily a concern. The concern is that the AI pipeline/orchestrator is the same and should return essentially the same answer each time.

Reply:
AZURE   (A  sure)   I   want  to  A  a or A A SURE    (Please   Close   Door)
No  want   publish  (Please  Publish our)     personal   porsonal